### ISSUE TITLE
Escaped character references (&amp;) are not preserved

### ISSUE THREAD

**Issue by JaredNeil** [stringham/angular-template-formatter#12]
```html
<div attr="&quot;">&lt;</div>
```
becomes
```html
<div attr="""><</div>
```

When fixing this we need to be careful to make sure stuff like this still works.
```html
<div [attr]="a < 2 ? '&quot;' : ''">{{ a && "&quot;" }}</div>
```

**Comment by shamrt** [comment]
In addition, other HTML entities, such as `&nbsp;` and `&#8287;` (regular space; occasionally required for our purposes) are removed.

**Comment by pauldraper** [comment]
@JaredNeil I assume your latter example was supposed to be

```html
<div [attr]="a < 2 ? '&quot;' : ''">{{ a && "&quot;" }}</div>
```

though that results in a Angular parse error anyway when it parses `a && """` as JS. Angular first uses an HTML parser and then its pseudo-JS parser. (I.e. there's no cdata or `<script>` type exit from the HTML parser.)

**Comment by alexw10** [comment]
Any update on this? 
Major issue that affects our project preventing us from utilizing this tool.

**Comment by stringham** [comment]
I'm accepting pull requests for this issue if anyone wants to take a look.

**Comment by stringham** [comment]
Fixed in version 0.1.2
